contracts/SwapKiwiV2.sol

```diff
@@ -0,0 +1,320 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.1;
+
+import "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol";
+import "@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol";
+import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract SwapKiwiV2 is Ownable, ERC721Holder, ERC1155Holder {
+
+    uint256 private _swapsCounter;
+    uint256 private _etherLocked;
+
+    uint256 public fee;
```

Comment:
I think if we reorder and use smaller types for _swapsCounter, _etherLocked and fee we could use less storage

Reply:
make sense, 
The type of _etherLocked should be the same as initiatorEtherValue and secondUserEtherValue.
So it should be uint128.
And also I think the type of fee should be uint128.
And uint64 is enough for the type of _swapsCounter.

Reply:
Yeah think so too. but maybe you should add tests first so we can monitor how much those changes impact gas costs?

Reply:
yeah